src/index.ts

```diff
@@ -1,178 +1,272 @@
-// Generated data
-import {
-  GPU_BENCHMARK_SCORE_DESKTOP,
-  GPU_BENCHMARK_SCORE_MOBILE,
-} from './__generated__/GPUBenchmark';
+// Vendor
+import leven from 'leven';
+import fetch from 'unfetch';
 
 // Internal
-import { cleanEntryString } from './internal/cleanEntryString';
-import { cleanRendererString } from './internal/cleanRendererString';
-import { getBenchmarkByPercentage } from './internal/getBenchmarkByPercentage';
-import { browser, isMobile, isTablet } from './internal/getBrowserType';
-import { getEntryVersionNumber } from './internal/getEntryVersionNumber';
-import { getWebGLUnmaskedRenderer } from './internal/getWebGLUnmaskedRenderer';
-import { isWebGLSupported } from './internal/isWebGLSupported';
-import { getLevenshteinDistance } from './internal/getLevenshteinDistance';
+import { getGPUVersion } from './internal/getGPUVersion';
+import { getWebGLContext } from './internal/getWebGLContext';
+import { deviceInfo, isSSR } from './internal/deviceInfo';
+import { deobfuscateRenderer } from './internal/deobfuscateRenderer';
 
 // Types
-export interface IGetGPUTier {
-  glContext?: WebGLRenderingContext | WebGL2RenderingContext;
-  mobileBenchmarkPercentages?: number[];
-  desktopBenchmarkPercentages?: number[];
-  failIfMajorPerformanceCaveat?: boolean;
-  forceRendererString?: string;
-  forceMobile?: boolean;
-}
-
-export type TRank = [number, string] | [undefined, undefined];
-
-export interface IRankWithDistance {
-  rank: TRank;
-  distance: number;
-}
-
-export const getGPUTier = ({
-  mobileBenchmarkPercentages = [
-    0, // TIER_0
-    50, // TIER_1
-    30, // TIER_2
-    20, // TIER_3
-  ],
-  desktopBenchmarkPercentages = [
-    0, // TIER_0
-    50, // TIER_1
-    30, // TIER_2
-    20, // TIER_3
-  ],
-  forceRendererString = '',
-  forceMobile = false,
+import type { IGetGPUTier, TModelEntry, TTierResult, TTierType } from './types';
```

Comment:
@puckey why do we use the `import type` syntax?

Reply:
https://www.typescriptlang.org/docs/handbook/modules.html#importing-types